### ISSUE TITLE
Encoding url parameters that are themselves urls

### ISSUE THREAD

**Issue by amal-qokka** [softwaremill/sttp#191]
I'm trying to perform form submission w/ sttp - one of the parameters in the form is a url suffix with query parameters.

When the query fragment containing the url suffix is encoded the url suffix is not completely encoded by Uri.QueryFragmentEncoding.Standard

I can specify the encoding by calling queryFragment on the uri object with encoding Uri.QueryFragmentEncoding.All:
```
val postUri = formParams.foldLeft(uri"$formHost"){case (u,(k,v)) => u.queryFragment(KeyValue(k,v,Uri.QueryFragmentEncoding.All,Uri.QueryFragmentEncoding.All))}
```
This works, but is clunky and it took me a while to figure out.

Is there a way to avoid using the Uri class and submit a string to request constructors?  Alternatively, is there a way to construct a Uri w/ a plain string w/ no interpolation, parsing, encoding, or decoding?

**Comment by adamw** [comment]
I'm not sure what you mean by "is not completely encoded"? `Uri.QueryFragmentEncoding.Standard` encodes query parameters as specified in https://tools.ietf.org/html/rfc3986

As for creating a request using a literal URI - that's not currently possible. The URI is always parsed. The closest you can get is `uri"$uriAsString`

**Comment by adamw** [comment]
Fixed in 2.0.0-M6: URI encoding is fully configurable
